grin/src/AbstractInterpretation/EffectTracking/CodeGenBase.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE LambdaCase, RecordWildCards, RankNTypes, TemplateHaskell #-}
+module AbstractInterpretation.EffectTracking.CodeGenBase where
+
+import Data.Int
+import Data.Word
+import Data.Set (Set)
+import Data.Map (Map)
+import Data.Vector (Vector)
+
+import qualified Data.Bimap as Bimap
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Data.Vector as Vec
+
+import Data.Functor.Infix
+
+import Control.Monad.State
+
+import Grin.Grin (Name, SimpleType(..), CPat(..), unpackName, Tag(..), External(..))
+import Grin.TypeEnvDefs
+import AbstractInterpretation.IR (Instruction(..), Reg(..), AbstractMapping)
+import qualified AbstractInterpretation.IR as IR
+
+import Lens.Micro.Platform
+
+type ExternalID = Int32
+
+data External' = E { extID  :: ExternalID
```

Comment:
I am strongly against tick in names even if it for local data, because they are slippery slope. Please could you find a better name here?

Reply:
Also if you find one from me, hit me with a stick ;)

Reply:
Was thinking hard, but didn't find anything descriptive. Now I settled on `ExternalWithID`.

Reply:
This is way better than the tick :)